LoRaEngine/modules/LoRaWanNetworkSrvModule/LoRaWan.NetworkServer/LoRaDeviceCache.cs

```diff
@@ -105,10 +105,11 @@ private async Task RefreshCacheAsync(CancellationToken cancellationToken)
             }
         }
 
-        protected virtual Task RefreshDeviceAsync(LoRaDevice device, CancellationToken cancellationToken)
+        protected virtual async Task RefreshDeviceAsync(LoRaDevice device, CancellationToken cancellationToken)
         {
             _ = device ?? throw new ArgumentNullException(nameof(device));
-            return device.InitializeAsync(this.configuration, cancellationToken);
+            using var activity = device.BeginDeviceClientConnectionActivity();
+            _ = await device.InitializeAsync(this.configuration, cancellationToken);
```

Comment:
nit: since `activity` is unused and this `using` form doesn't allow discards, how about just going back to the good old `using` block with no variable assignment?

```suggestion
            using (device.BeginDeviceClientConnectionActivity())
                _ = await device.InitializeAsync(this.configuration, cancellationToken);
```